### ISSUE TITLE
would genereated text be different for different gpus A100(40GB) vs A10G(24GB) for same finetuned model

### ISSUE THREAD

**Issue by bhupendrathore** [vllm-project/vllm#1549]
TLDR : vllm is great. though i have a question i have finetuned a mistral on A100 40GB (QLora) and merged the model. I see the model is quite consitent on A100 but when i infer the same model on A10G (24GB), with same parameter the model output text is inconsistent mostly. dependencies are same. but cuda version is 11.7 vs cuda 12.2 . i am using dtype = bfloat16 for both and trained with bfloat.

any suggestions ... i might be doing something wrong or it's normal !!!  also when i use inference api using openai.api_server output is little different for same parameters.. i understand that mostly due to randomness.

Thanks

**Comment by LiuXiaoxuanPKU** [comment]
Do you see the difference by using greedy sampling (i.e., setting temperature to 0)?

**Comment by lijianxing123** [comment]
I have the same problem in A100 and A40, does anyone have a solution?
